rpcserver.go

```diff
@@ -936,12 +941,118 @@ func (r *rpcServer) VerifyProof(ctx context.Context,
 	)
 	valid := err == nil
 
-	// TODO(roasbeef): also show additional final resting anchor
-	// information, etc?
+	decodedProof, err := marshalProofFile(proofFile, -1)
+	if err != nil {
+		return nil, fmt.Errorf("unable to marshal proof: %w", err)
+	}
+
+	return &tarorpc.VerifyProofResponse{
+		Valid:        valid,
+		DecodedProof: decodedProof,
+	}, nil
+}
+
+// DecodeProof attempts to decode a given proof file that claims to be anchored
+// at the specified genesis point.
+func (r *rpcServer) DecodeProof(ctx context.Context,
+	in *tarorpc.DecodeProofRequest) (*tarorpc.DecodeProofResponse, error) {
+
+	if len(in.RawProof) == 0 {
+		return nil, fmt.Errorf("proof file must be specified")
+	}
+
+	var proofFile proof.File
+	err := proofFile.Decode(bytes.NewReader(in.RawProof))
+	if err != nil {
+		return nil, fmt.Errorf("unable to decode proof file: %w", err)
+	}
+
+	decodedProof, err := marshalProofFile(proofFile, in.ProofIndex)
+	if err != nil {
+		return nil, fmt.Errorf("unable to marshal proof: %w", err)
+	}
+
+	return &tarorpc.DecodeProofResponse{
+		DecodedProof: decodedProof,
+	}, nil
+}
+
+// marshalProofFile turns a proof file into an RPC DecodedProof.
+func marshalProofFile(proofFile proof.File,
+	index int64) (*tarorpc.DecodedProof, error) {
+	var (
+		decodedProof *proof.Proof
+		err          error
+	)
+
+	// Default to latest proof
+	if index < 0 {
+		decodedProof, err = proofFile.LastProof()
+		if err != nil {
+			return nil, err
+		}
+		index = int64(proofFile.NumProofs() - 1)
+	} else {
+		decodedProof, err = proofFile.ProofAt(uint32(index))
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	var (
+		finalAsset     = decodedProof.Asset
+		rpcAsset       = &tarorpc.Asset{}
+		anchorOutpoint = wire.OutPoint{
+			decodedProof.AnchorTx.TxHash(),
+			decodedProof.InclusionProof.OutputIndex,
+		}
+	)
+
+	var txMerkleProofBuf bytes.Buffer
+	if err := decodedProof.TxMerkleProof.Encode(&txMerkleProofBuf); err != nil {
+		return nil, fmt.Errorf("unable to encode serialized Bitcoin merkle proof: %w",
```

Comment:
nit: line length (remember to set your IDE to use 8 spaces for the tab character).